cezo_fl/util/prepare_settings.py

```diff
@@ -0,0 +1,234 @@
+from typing import Literal, Callable
+import torch
+import torch.nn as nn
+
+from functools import partial
+from peft import LoraConfig, get_peft_model
+from transformers import AutoModelForCausalLM
+
+
+from cezo_fl.util import model_helpers
+from cezo_fl.util.model_helpers import AllModel
+from cezo_fl.models.cnn_fashion import CNN_FMNIST
+from cezo_fl.models.cnn_mnist import CNN_MNIST
+from cezo_fl.models.lenet import LeNet
+from cezo_fl.models.lstm import CharLSTM
+from cezo_fl.random_gradient_estimator import RandomGradientEstimator as RGE
+from cezo_fl.util.language_utils import (
+    LM_TEMPLATE_MAP,
+    SUPPORTED_LLM,
+    get_lm_loss,
+    get_hf_tokenizer,
+)
+from cezo_fl.util.metrics import accuracy
+
+from dataclasses import dataclass
+
+
+def get_torch_dtype(args_model_dtype: Literal["float32", "float16", "bfloat16"]) -> torch.dtype:
+    return {
+        "float32": torch.float32,
+        "float16": torch.float16,
+        "bfloat16": torch.bfloat16,
+    }[args_model_dtype]
+
+
+def get_random_gradient_estimator(args, model, device):
```

Comment:
nit: type annotation

Reply:
Optional: this is also a good start point to decouple it from `args`:

```python
@dataclass
class RGE_Params:
     mu: float
     num_pert: int
     grad_estimate_method: str
     model_dtype: torch.dtype
     sgd_only_no_optim: bool
     paramwise_perturb: bool

     def parse_from_args():
          ....

def get_random_gradient_estimator(rge_param: RGE_Params, model, device):
        ....
```
This RGE_Params dataclass can be used in the RGE as well to avoid misalignment

Reply:
> Optional: this is also a good start point to decouple it from `args`:
> 
> ```python
> @dataclass
> class RGE_Params:
>      mu: float
>      num_pert: int
>      grad_estimate_method: str
>      model_dtype: torch.dtype
>      sgd_only_no_optim: bool
>      paramwise_perturb: bool
> 
>      def parse_from_args():
>           ....
> 
> def get_random_gradient_estimator(rge_param: RGE_Params, model, device):
>         ....
> ```
> 
> This RGE_Params dataclass can be used in the RGE as well to avoid misalignment

this is great suggestion! I will implement this change for all cli parameter groups in a later PR.